### DISCUSSION TITLE
Column value undefined in table with conditional columns

### DISCUSSION THREAD

**Discussion by rourke** [Budibase/budibase#5631] [Help]
I have a Data Provider with a table in it. When a checkbox is true it should include an extra column. So the table has no columns configured. Instead the columns are configured in the Table's conditions. This means for both checkbox conditions the table column conditions are configured. This works fine.

But when I change the value of the the columns I can't manipulate the value through handlebars or Javascript.

![image](https://user-images.githubusercontent.com/2820753/165738864-ef4e2980-a499-4546-b267-ca288ff4f8bf.png)

![image](https://user-images.githubusercontent.com/2820753/165737171-56efa6a2-383f-45b6-a9e8-43d231c1ba0c.png)

A basic multiply fails
![image](https://user-images.githubusercontent.com/2820753/165737322-7c4db24b-8d60-4f0b-8d6e-4af055934b56.png)

`$("Value")` returns undefined in the Javascript editor.
![image](https://user-images.githubusercontent.com/2820753/165737724-09c74404-44db-4f46-b6ea-8bc0fd50453a.png)

With the above Javascript the amount column looks like this:
![image](https://user-images.githubusercontent.com/2820753/165738428-be6652c0-7caa-46d8-bf23-ab544841ef45.png)

Am I doing something wrong or is this some sort of bug?

**Comment by melohagan** [comment]
Hi @rourke 

This may be related to this bug here: https://github.com/Budibase/budibase/issues/5365

If you keep the two `Update Setting` but include the additional column in both cases, is the value defined then? 

It's possible that it is trying to do the binding before it has shown the extra column.

**Reply by rourke** [reply]
The column I'm showing/hiding is a different column from the `Bedrag` column that has trouble displaying data from my example. But I've added the column in both `Update Setting`. Still has the same issue.

![image](https://user-images.githubusercontent.com/2820753/165783365-37eacb48-bd41-4d9f-956e-595940f968d2.png)

![image](https://user-images.githubusercontent.com/2820753/165783674-36195d84-2122-47af-a540-d971003a8e20.png)

**Reply by rourke** [reply]
I love dynamic components showing only the information I need. So I make heavily use of conditions and this problem is something I run into a lot.

I think this can be considered a bug? Should I make a separate bug issue for this?

**Reply by melohagan** [reply]
Hey @rourke 

Sorry I linked the wrong issue, I meant this one here (updated the OP as well): https://github.com/Budibase/budibase/issues/5365

**Reply by rourke** [reply]
Thanks @melohagan. I'm not familiar enough with the logic to understand if this is the same issue. But it looks similar.

For now I managed to create a workaround by adding a formula column in the database table that does the formatting. When the PR is merged and live I will see if my issue is resolved.
